Compile error if we only use VCS without VOCS and AICS

**Describe the bug**
There will be compile error if we only enabled the VCS through
```
CONFIG_BT_VCS=y
CONFIG_BT_VCS_CLIENT=y
```


**To Reproduce**
We can use "zephyr/samples/bluetooth/peripheral_hr" for replicating the issue
1. Add following configs in the prj.conf
```
CONFIG_BT_AUDIO=y
CONFIG_BT_VCS=y
CONFIG_BT_VCS_CLIENT=y
```
2. Compile the project through west 
```
west build -b nrf5340dk_nrf5340_cpuapp -p
```
3. And then we will the see the error log like
```
In file included from /home/rick/zephyr/subsys/bluetooth/audio/vcs_client.c:13:
/home/rick/zephyr/subsys/bluetooth/audio/vcs_client.c: In function 'bt_vcs_client_cb_register':
/home/rick/zephyr/subsys/bluetooth/audio/vcs_client.c:790:23: error: 'struct bt_vocs_cb' has no member named 'discover'
  790 |    CHECKIF(cb->vocs_cb.discover) {
      |                       ^
/home/rick/zephyr/include/sys/check.h:22:6: note: in definition of macro 'CHECKIF'
   22 |  if (expr)
      |      ^~~~
/home/rick/zephyr/subsys/bluetooth/audio/vcs_client.c:794:15: error: 'struct bt_vocs_cb' has no member named 'discover'
  794 |    cb->vocs_cb.discover = vocs_discover_cb;
      |               ^
In file included from /home/rick/zephyr/subsys/bluetooth/audio/vcs_client.c:13:
/home/rick/zephyr/subsys/bluetooth/audio/vcs_client.c:813:23: error: 'struct bt_aics_cb' has no member named 'discover'
  813 |    CHECKIF(cb->aics_cb.discover) {
      |                       ^
/home/rick/zephyr/include/sys/check.h:22:6: note: in definition of macro 'CHECKIF'
   22 |  if (expr)
      |      ^~~~
/home/rick/zephyr/subsys/bluetooth/audio/vcs_client.c:817:15: error: 'struct bt_aics_cb' has no member named 'discover'
  817 |    cb->aics_cb.discover = aics_discover_cb;
      |               ^
```

**Expected behavior**
There should be no issue for using only VCS without enabling the AICS and VOCS, which is the default setting if we only enabled ```CONFIG_BT_VCS```  and ```CONFIG_BT_VCS_CLIENT```

**Impact**
We won't be able to use the minimal configuration for VCS which without VOCS and AICS.

**Environment (please complete the following information):**
 - OS: Linux 64bits
 - Toolchain: GNU Arm Embed 10.2.1
 - Commit SHA: f3d9808012d24885b9e4051981ca5d5afe2e7b4a

--- a/subsys/bluetooth/audio/aics.c
+++ b/subsys/bluetooth/audio/aics.c
@@ -29,6 +29,18 @@
 #define AICS_CP_LEN                 0x02
 #define AICS_CP_SET_GAIN_LEN        0x03
 
+
+static ssize_t write_description(struct bt_conn *conn,
+				 const struct bt_gatt_attr *attr,
+				 const void *buf, uint16_t len, uint16_t offset,
+				 uint8_t flags);
+
+static ssize_t write_aics_control(struct bt_conn *conn,
+				  const struct bt_gatt_attr *attr,
+				  const void *buf, uint16_t len,
+				  uint16_t offset, uint8_t flags);
+
+#if defined(CONFIG_BT_AICS)
 static void aics_state_cfg_changed(const struct bt_gatt_attr *attr,
 				   uint16_t value);
 static ssize_t read_aics_state(struct bt_conn *conn,
@@ -45,21 +57,12 @@ static void aics_input_status_cfg_changed(const struct bt_gatt_attr *attr,
 static ssize_t read_input_status(struct bt_conn *conn,
 				 const struct bt_gatt_attr *attr,
 				 void *buf, uint16_t len, uint16_t offset);
-static ssize_t write_aics_control(struct bt_conn *conn,
-				  const struct bt_gatt_attr *attr,
-				  const void *buf, uint16_t len,
-				  uint16_t offset, uint8_t flags);
 static void aics_description_cfg_changed(const struct bt_gatt_attr *attr,
 					 uint16_t value);
-static ssize_t write_description(struct bt_conn *conn,
-				 const struct bt_gatt_attr *attr,
-				 const void *buf, uint16_t len, uint16_t offset,
-				 uint8_t flags);
 static ssize_t read_description(struct bt_conn *conn,
 				const struct bt_gatt_attr *attr, void *buf,
 				uint16_t len, uint16_t offset);
 
-#if defined(CONFIG_BT_AICS)
 #define BT_AICS_SERVICE_DEFINITION(_aics) {                                    \
 	BT_GATT_SECONDARY_SERVICE(BT_UUID_AICS),                               \
 	BT_GATT_CHARACTERISTIC(BT_UUID_AICS_STATE,                             \

--- a/subsys/bluetooth/audio/mics_client.c
+++ b/subsys/bluetooth/audio/mics_client.c
@@ -32,21 +32,6 @@ static struct bt_mics_cb *mics_client_cb;
 static struct bt_mics mics_insts[CONFIG_BT_MAX_CONN];
 static struct bt_uuid *mics_uuid = BT_UUID_MICS;
 
-static struct bt_mics *lookup_mics_by_aics(const struct bt_aics *aics)
-{
-	__ASSERT(aics != NULL, "AICS pointer cannot be NULL");
-
-	for (int i = 0; i < ARRAY_SIZE(mics_insts); i++) {
-		for (int j = 0; j < ARRAY_SIZE(mics_insts[i].cli.aics); j++) {
-			if (mics_insts[i].cli.aics[j] == aics) {
-				return &mics_insts[i];
-			}
-		}
-	}
-
-	return NULL;
-}
-
 bool bt_mics_client_valid_aics_inst(struct bt_mics *mics, struct bt_aics *aics)
 {
 	if (mics == NULL) {
@@ -149,6 +134,22 @@ static void mics_client_write_mics_mute_cb(struct bt_conn *conn, uint8_t err,
 	}
 }
 
+#if defined(CONFIG_BT_MICS_CLIENT_AICS)
+static struct bt_mics *lookup_mics_by_aics(const struct bt_aics *aics)
+{
+	__ASSERT(aics != NULL, "AICS pointer cannot be NULL");
+
+	for (int i = 0; i < ARRAY_SIZE(mics_insts); i++) {
+		for (int j = 0; j < ARRAY_SIZE(mics_insts[i].cli.aics); j++) {
+			if (mics_insts[i].cli.aics[j] == aics) {
+				return &mics_insts[i];
+			}
+		}
+	}
+
+	return NULL;
+}
+
 static void aics_discover_cb(struct bt_aics *inst, int err)
 {
 	struct bt_mics *mics_inst = lookup_mics_by_aics(inst);
@@ -167,6 +168,7 @@ static void aics_discover_cb(struct bt_aics *inst, int err)
 		}
 	}
 }
+#endif /* CONFIG_BT_MICS_CLIENT_AICS */
 
 static uint8_t mics_discover_include_func(
 	struct bt_conn *conn, const struct bt_gatt_attr *attr,
@@ -429,30 +431,29 @@ int bt_mics_discover(struct bt_conn *conn, struct bt_mics **mics)
 
 int bt_mics_client_cb_register(struct bt_mics_cb *cb)
 {
-	int i, j;
+#if defined(CONFIG_BT_MICS_CLIENT_AICS)
+	struct bt_aics_cb *aics_cb = NULL;
 
-	if (IS_ENABLED(CONFIG_BT_AICS_CLIENT)) {
-		struct bt_aics_cb *aics_cb = NULL;
-
-		if (cb != NULL) {
-			CHECKIF(cb->aics_cb.discover != NULL) {
-				BT_ERR("AICS discover callback shall not be set");
-				return -EINVAL;
-			}
-			cb->aics_cb.discover = aics_discover_cb;
-
-			aics_cb = &cb->aics_cb;
+	if (cb != NULL) {
+		CHECKIF(cb->aics_cb.discover != NULL) {
+			BT_ERR("AICS discover callback shall not be set");
+			return -EINVAL;
 		}
+		cb->aics_cb.discover = aics_discover_cb;
 
-		for (i = 0; i < ARRAY_SIZE(mics_insts); i++) {
-			for (j = 0; j < ARRAY_SIZE(mics_insts[i].cli.aics); j++) {
-				if (mics_insts[i].cli.aics[j] != NULL) {
-					bt_aics_client_cb_register(mics_insts[i].cli.aics[j],
-								   aics_cb);
-				}
+		aics_cb = &cb->aics_cb;
+	}
+
+	for (int i = 0; i < ARRAY_SIZE(mics_insts); i++) {
+		for (int j = 0; j < ARRAY_SIZE(mics_insts[i].cli.aics); j++) {
+			struct bt_aics *aics = mics_insts[i].cli.aics[j];
+
+			if (aics != NULL) {
+				bt_aics_client_cb_register(aics, aics_cb);
 			}
 		}
 	}
+#endif /* CONFIG_BT_MICS_CLIENT_AICS */
 
 	mics_client_cb = cb;
 

--- a/subsys/bluetooth/audio/mics_internal.h
+++ b/subsys/bluetooth/audio/mics_internal.h
@@ -9,13 +9,16 @@
 #include <zephyr/types.h>
 #include <bluetooth/gatt.h>
 
+#if defined(CONFIG_BT_MICS)
 struct bt_mics_server {
 	uint8_t mute;
 	struct bt_mics_cb *cb;
 	struct bt_gatt_service *service_p;
 	struct bt_aics *aics_insts[CONFIG_BT_MICS_AICS_INSTANCE_COUNT];
 };
+#endif /* CONFIG_BT_MICS */
 
+#if defined(CONFIG_BT_MICS_CLIENT)
 struct bt_mics_client {
 	uint16_t start_handle;
 	uint16_t end_handle;
@@ -33,13 +36,18 @@ struct bt_mics_client {
 	uint8_t aics_inst_cnt;
 	struct bt_aics *aics[CONFIG_BT_MICS_CLIENT_MAX_AICS_INST];
 };
+#endif /* CONFIG_BT_MICS_CLIENT */
 
 /* Struct used as a common type for the api */
 struct bt_mics {
 	bool client_instance;
 	union {
+#if defined(CONFIG_BT_MICS)
 		struct bt_mics_server srv;
+#endif /* CONFIG_BT_MICS */
+#if defined(CONFIG_BT_MICS_CLIENT)
 		struct bt_mics_client cli;
+#endif /* CONFIG_BT_MICS_CLIENT */
 	};
 };
 

--- a/subsys/bluetooth/audio/vcs_client.c
+++ b/subsys/bluetooth/audio/vcs_client.c
@@ -32,36 +32,6 @@ static struct bt_vcs_cb *vcs_client_cb;
 static struct bt_vcs vcs_insts[CONFIG_BT_MAX_CONN];
 static int vcs_client_common_vcs_cp(struct bt_vcs *vcs, uint8_t opcode);
 
-static struct bt_vcs *lookup_vcs_by_vocs(const struct bt_vocs *vocs)
-{
-	__ASSERT(vocs != NULL, "VOCS pointer cannot be NULL");
-
-	for (int i = 0; i < ARRAY_SIZE(vcs_insts); i++) {
-		for (int j = 0; j < ARRAY_SIZE(vcs_insts[i].cli.vocs); j++) {
-			if (vcs_insts[i].cli.vocs[j] == vocs) {
-				return &vcs_insts[i];
-			}
-		}
-	}
-
-	return NULL;
-}
-
-static struct bt_vcs *lookup_vcs_by_aics(const struct bt_aics *aics)
-{
-	__ASSERT(aics != NULL, "aics pointer cannot be NULL");
-
-	for (int i = 0; i < ARRAY_SIZE(vcs_insts); i++) {
-		for (int j = 0; j < ARRAY_SIZE(vcs_insts[i].cli.aics); j++) {
-			if (vcs_insts[i].cli.aics[j] == aics) {
-				return &vcs_insts[i];
-			}
-		}
-	}
-
-	return NULL;
-}
-
 bool bt_vcs_client_valid_vocs_inst(struct bt_vcs *vcs, struct bt_vocs *vocs)
 {
 	if (vcs == NULL) {
@@ -624,6 +594,22 @@ static int vcs_client_common_vcs_cp(struct bt_vcs *vcs, uint8_t opcode)
 	return err;
 }
 
+#if defined(CONFIG_BT_VCS_CLIENT_AICS)
+static struct bt_vcs *lookup_vcs_by_aics(const struct bt_aics *aics)
+{
+	__ASSERT(aics != NULL, "aics pointer cannot be NULL");
+
+	for (int i = 0; i < ARRAY_SIZE(vcs_insts); i++) {
+		for (int j = 0; j < ARRAY_SIZE(vcs_insts[i].cli.aics); j++) {
+			if (vcs_insts[i].cli.aics[j] == aics) {
+				return &vcs_insts[i];
+			}
+		}
+	}
+
+	return NULL;
+}
+
 static void aics_discover_cb(struct bt_aics *inst, int err)
 {
 	struct bt_vcs *vcs_inst = lookup_vcs_by_aics(inst);
@@ -640,6 +626,23 @@ static void aics_discover_cb(struct bt_aics *inst, int err)
 			vcs_client_cb->discover(vcs_inst, err, 0, 0);
 		}
 	}
+}
+#endif /* CONFIG_BT_VCS_CLIENT_AICS */
+
+#if defined(CONFIG_BT_VCS_CLIENT_VOCS)
+static struct bt_vcs *lookup_vcs_by_vocs(const struct bt_vocs *vocs)
+{
+	__ASSERT(vocs != NULL, "VOCS pointer cannot be NULL");
+
+	for (int i = 0; i < ARRAY_SIZE(vcs_insts); i++) {
+		for (int j = 0; j < ARRAY_SIZE(vcs_insts[i].cli.vocs); j++) {
+			if (vcs_insts[i].cli.vocs[j] == vocs) {
+				return &vcs_insts[i];
+			}
+		}
+	}
+
+	return NULL;
 }
 
 static void vocs_discover_cb(struct bt_vocs *inst, int err)
@@ -671,6 +674,7 @@ static void vocs_discover_cb(struct bt_vocs *inst, int err)
 		}
 	}
 }
+#endif /* CONFIG_BT_VCS_CLIENT_VOCS */
 
 static void vcs_client_reset(struct bt_conn *conn)
 {
@@ -781,53 +785,53 @@ int bt_vcs_discover(struct bt_conn *conn, struct bt_vcs **vcs)
 
 int bt_vcs_client_cb_register(struct bt_vcs_cb *cb)
 {
-	int i, j;
+#if defined(CONFIG_BT_VCS_CLIENT_VOCS)
+	struct bt_vocs_cb *vocs_cb = NULL;
 
-	if (IS_ENABLED(CONFIG_BT_VOCS_CLIENT)) {
-		struct bt_vocs_cb *vocs_cb = NULL;
-
-		if (cb != NULL) {
-			CHECKIF(cb->vocs_cb.discover) {
-				BT_ERR("VOCS discover callback shall not be set");
-				return -EINVAL;
-			}
-			cb->vocs_cb.discover = vocs_discover_cb;
-
-			vocs_cb = &cb->vocs_cb;
+	if (cb != NULL) {
+		CHECKIF(cb->vocs_cb.discover) {
+			BT_ERR("VOCS discover callback shall not be set");
+			return -EINVAL;
 		}
+		cb->vocs_cb.discover = vocs_discover_cb;
 
-		for (i = 0; i < ARRAY_SIZE(vcs_insts); i++) {
-			for (j = 0; j < ARRAY_SIZE(vcs_insts[i].cli.vocs); j++) {
-				if (vcs_insts[i].cli.vocs[j]) {
-					bt_vocs_client_cb_register(vcs_insts[i].cli.vocs[j],
-								   vocs_cb);
-				}
+		vocs_cb = &cb->vocs_cb;
+	}
+
+	for (int i = 0; i < ARRAY_SIZE(vcs_insts); i++) {
+		for (int j = 0; j < ARRAY_SIZE(vcs_insts[i].cli.vocs); j++) {
+			struct bt_vocs *vocs = vcs_insts[i].cli.vocs[j];
+
+			if (vocs != NULL) {
+				bt_vocs_client_cb_register(vocs, vocs_cb);
 			}
 		}
 	}
+#endif /* CONFIG_BT_VCS_CLIENT_VOCS */
 
-	if (IS_ENABLED(CONFIG_BT_AICS_CLIENT)) {
-		struct bt_aics_cb *aics_cb = NULL;
+#if defined(CONFIG_BT_VCS_CLIENT_AICS)
+	struct bt_aics_cb *aics_cb = NULL;
 
-		if (cb != NULL) {
-			CHECKIF(cb->aics_cb.discover) {
-				BT_ERR("AICS discover callback shall not be set");
-				return -EINVAL;
-			}
-			cb->aics_cb.discover = aics_discover_cb;
-
-			aics_cb = &cb->aics_cb;
+	if (cb != NULL) {
+		CHECKIF(cb->aics_cb.discover) {
+			BT_ERR("AICS discover callback shall not be set");
+			return -EINVAL;
 		}
+		cb->aics_cb.discover = aics_discover_cb;
 
-		for (i = 0; i < ARRAY_SIZE(vcs_insts); i++) {
-			for (j = 0; j < ARRAY_SIZE(vcs_insts[i].cli.aics); j++) {
-				if (vcs_insts[i].cli.aics[j]) {
-					bt_aics_client_cb_register(vcs_insts[i].cli.aics[j],
-								   aics_cb);
-				}
+		aics_cb = &cb->aics_cb;
+	}
+
+	for (int i = 0; i < ARRAY_SIZE(vcs_insts); i++) {
+		for (int j = 0; j < ARRAY_SIZE(vcs_insts[i].cli.aics); j++) {
+			struct bt_aics *aics = vcs_insts[i].cli.aics[j];
+
+			if (aics != NULL) {
+				bt_aics_client_cb_register(aics, aics_cb);
 			}
 		}
 	}
+#endif /* CONFIG_BT_VCS_CLIENT_AICS */
 
 	vcs_client_cb = cb;
 

--- a/subsys/bluetooth/audio/vcs_internal.h
+++ b/subsys/bluetooth/audio/vcs_internal.h
@@ -36,6 +36,7 @@ struct vcs_control_vol {
 	uint8_t volume;
 } __packed;
 
+#if defined(CONFIG_BT_VCS_CLIENT)
 struct bt_vcs_client {
 	struct vcs_state state;
 	uint8_t flags;
@@ -62,7 +63,9 @@ struct bt_vcs_client {
 	uint8_t aics_inst_cnt;
 	struct bt_aics *aics[CONFIG_BT_VCS_CLIENT_MAX_AICS_INST];
 };
+#endif /* CONFIG_BT_VCS_CLIENT */
 
+#if defined(CONFIG_BT_VCS)
 struct bt_vcs_server {
 	struct vcs_state state;
 	uint8_t flags;
@@ -73,13 +76,18 @@ struct bt_vcs_server {
 	struct bt_vocs *vocs_insts[CONFIG_BT_VCS_VOCS_INSTANCE_COUNT];
 	struct bt_aics *aics_insts[CONFIG_BT_VCS_AICS_INSTANCE_COUNT];
 };
+#endif /* CONFIG_BT_VCS */
 
 /* Struct used as a common type for the api */
 struct bt_vcs {
 	bool client_instance;
 	union {
+#if defined(CONFIG_BT_VCS)
 		struct bt_vcs_server srv;
+#endif /* CONFIG_BT_VCS */
+#if defined(CONFIG_BT_VCS_CLIENT)
 		struct bt_vcs_client cli;
+#endif /* CONFIG_BT_VCS_CLIENT */
 	};
 };
 

--- a/tests/bluetooth/shell/testcase.yaml
+++ b/tests/bluetooth/shell/testcase.yaml
@@ -28,3 +28,86 @@ tests:
     extra_args: CONFIG_BT_PRIVACY=n
     platform_allow: native_posix
     tags: bluetooth
+
+# Bluetooth Audio Compile validation tests
+  bluetooth.shell.no_vcs:
+    build_only: true
+    integration_platforms:
+      - native_posix
+    extra_configs:
+      - CONFIG_BT_VCS=n
+  bluetooth.shell.no_vocs:
+    build_only: true
+    integration_platforms:
+      - native_posix
+    extra_configs:
+      - CONFIG_BT_VOCS_MAX_INSTANCE_COUNT=0
+      - CONFIG_BT_VCS_VOCS_INSTANCE_COUNT=0
+  bluetooth.shell.no_aics:
+    build_only: true
+    integration_platforms:
+      - native_posix
+    extra_configs:
+      - CONFIG_BT_AICS_MAX_INSTANCE_COUNT=0
+      - CONFIG_BT_VCS_AICS_INSTANCE_COUNT=0
+      - CONFIG_BT_MICS_AICS_INSTANCE_COUNT=0
+  bluetooth.shell.no_aics_vocs:
+    build_only: true
+    integration_platforms:
+      - native_posix
+    extra_configs:
+      - CONFIG_BT_VOCS_MAX_INSTANCE_COUNT=0
+      - CONFIG_BT_VCS_VOCS_INSTANCE_COUNT=0
+      - CONFIG_BT_AICS_MAX_INSTANCE_COUNT=0
+      - CONFIG_BT_VCS_AICS_INSTANCE_COUNT=0
+      - CONFIG_BT_MICS_AICS_INSTANCE_COUNT=0
+  bluetooth.shell.no_vcs_client:
+    build_only: true
+    integration_platforms:
+      - native_posix
+    extra_configs:
+      - CONFIG_BT_VCS_CLIENT=n
+  bluetooth.shell.no_vcs_vcs_client:
+    build_only: true
+    integration_platforms:
+      - native_posix
+    extra_configs:
+      - CONFIG_BT_VCS=n
+      - CONFIG_BT_VCS_CLIENT=n
+  bluetooth.shell.vcs_client_no_aics_client:
+    build_only: true
+    integration_platforms:
+      - native_posix
+    extra_configs:
+      - CONFIG_BT_VCS_CLIENT_MAX_AICS_INST=0
+  bluetooth.shell.vcs_client_no_vocs_client:
+    build_only: true
+    integration_platforms:
+      - native_posix
+    extra_configs:
+      - CONFIG_BT_VCS_CLIENT_MAX_VOCS_INST=0
+  bluetooth.shell.no_mics:
+    build_only: true
+    integration_platforms:
+      - native_posix
+    extra_configs:
+      - CONFIG_BT_MICS=n
+  bluetooth.shell.no_mics_client:
+    build_only: true
+    integration_platforms:
+      - native_posix
+    extra_configs:
+      - CONFIG_BT_MICS_CLIENT=n
+  bluetooth.shell.no_mics_mics_client:
+    build_only: true
+    integration_platforms:
+      - native_posix
+    extra_configs:
+      - CONFIG_BT_MICS=n
+      - CONFIG_BT_MICS_CLIENT=n
+  bluetooth.shell.mics_client_no_aics_client:
+    build_only: true
+    integration_platforms:
+      - native_posix
+    extra_configs:
+      - CONFIG_BT_MICS_CLIENT_MAX_AICS_INST=0

